courses/api_test.py

```diff
@@ -73,7 +73,8 @@ def course():
 @pytest.fixture()
 def passed_grade_with_enrollment(user):
     """Fixture to produce a passed CourseRunGrade"""
-    course_run = CourseRunFactory.create()
+    course = CourseFactory.create()
```

Comment:
Ideally, the CourseRunFactory should create the Course object automatically using CourseFactory. Was there anything specific for this change?

Reply:
For this test the `page` attribute in course object is required which is created when we call CourseFactory the course created from CourseRunFactory the `page` attribute remains None (this is because many test cases got failed when `page` is created by CourseRunFactory). The `page` attribute was added few days ago as we discussed this issue earlier. Whenever we need `page` attribute for tests then course should be created from CourseFactory not from the CourseRunFactory. If we add `page` to course via CourseRunFactory many of other tests got failed.

Reply:
I feel this behavior is a little inconsistent because this gets confusing at this point, one of the options could have been that the default initialization of `page` variable in the course factory should be set to some default factory page and anyone not needing a page could create the page object like `CourseFactory.create(course__page=None)` Anyway that's not associates with this PR.

But there is a [course fixture](https://github.com/mitodl/mitxonline/blob/4667f071687b999a6c33fa7fe114ffa882954b9a/courses/api_test.py#L69) that you can use instead of creating it here.

Reply:
fixed by setting `CourseFactory.create(course__page=None)`